angular-chess/src/app/components/chess-board.component.ts

```diff
@@ -200,4 +210,24 @@ export class ChessBoardComponent implements OnInit {
 
     this.overlayPanel?.hide();
   }
+
+  /**
+   * Returns the corresponding PieceType based on the given pieceName.
+   * @param pieceName - The name of the piece.
+   * @returns The corresponding PieceType.
+   */
+  private getPieceType(pieceName: string) {
+    if (pieceName === PieceTypeRepresentation.QUEEN.valueOf()) {
```

Comment:
Is a switch case block not better?

Reply:
`
private getPieceType(pieceName: string) {
  switch(pieceName) {
    case PieceTypeRepresentation.QUEEN.valueOf():
      return PieceType.QUEEN;
    case PieceTypeRepresentation.ROOK.valueOf():
      return PieceType.ROOK;
    case PieceTypeRepresentation.BISHOP.valueOf():
      return PieceType.BISHOP;
    case PieceTypeRepresentation.KNIGHT.valueOf():
      return PieceType.KNIGHT;
    default:
      LoggingUtils.log(LogLevel.ERROR, () => `getPieceType unknown piece: ${pieceName}`);
      return PieceType.PAWN;
  }
}
`